RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSeedProducer.cc

```diff
@@ -0,0 +1,542 @@
+// -*- C++ -*-
+//
+// Package:    ElectronProducers
+// Class:      LowPtGsfElectronSeedProducer
+//
+/**\class LowPtGsfElectronSeedProducer RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSeedProducer.cc
+ Description: EDProducer of ElectronSeed objects
+ Implementation:
+     <Notes on implementation>
+*/
+// Original Author:  Robert Bainbridge
+
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/GeometryVector/interface/GlobalVector.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "DataFormats/ParticleFlowReco/interface/PFTrajectoryPoint.h"
+#include "DataFormats/TrackReco/interface/TrackBase.h"
+#include "FastSimulation/BaseParticlePropagator/interface/BaseParticlePropagator.h"
+#include "FastSimulation/Particle/interface/RawParticle.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSeedProducer.h"
+#include "RecoTracker/TransientTrackingRecHit/interface/TkClonerImpl.h"
+#include "RecoTracker/TransientTrackingRecHit/interface/TkTransientTrackingRecHitBuilder.h"
+#include "TrackingTools/PatternTools/interface/Trajectory.h"
+#include "TrackingTools/Records/interface/TransientRecHitRecord.h"
+#include "TrackingTools/TrajectoryParametrization/interface/GlobalTrajectoryParameters.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateOnSurface.h"
+#include "TMath.h"
+
+//////////////////////////////////////////////////////////////////////////////////////////
+//
+LowPtGsfElectronSeedProducer::LowPtGsfElectronSeedProducer( const edm::ParameterSet& conf, 
+							    const lowptgsfeleseed::HeavyObjectCache* ) :
+  field_(),
+  fitterPtr_(),
+  smootherPtr_(),
+  kfTracks_(),
+  pfTracks_(),
+  ecalClusters_{consumes<reco::PFClusterCollection>(conf.getParameter<edm::InputTag>("ecalClusters"))},
+  hcalClusters_(),
+  ebRecHits_{consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("EBRecHits"))},
+  eeRecHits_{consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("EERecHits"))},
+  rho_(consumes<double>(conf.getParameter<edm::InputTag>("rho"))),
```

Comment:
Is this "rho_" ever needed? It doesn't seem so.
This introduces a circular dependency, because you claim consuming some product which is only produced later in the chain